kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/net/BootstrapServerSelectionTest.java

```diff
@@ -0,0 +1,93 @@
+package io.kroxylicious.proxy.internal.net;
+
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import io.kroxylicious.proxy.service.HostPort;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test to verify that bootstrap server selection distributes across multiple servers
+ * instead of always selecting the first one.
+ */
+class BootstrapServerSelectionTest {
+    
+    @Test
+    void shouldSelectDifferentBootstrapServersOverTime() {
+        // Arrange
+        List<HostPort> bootstrapServers = List.of(
+            HostPort.parse("server1:9092"),
+            HostPort.parse("server2:9092"), 
+            HostPort.parse("server3:9092")
+        );
+        
+        EndpointRegistry registry = new EndpointRegistry(new TestNetworkBindingOperationProcessor());
+        Set<HostPort> selectedServers = new HashSet<>();
+        
+        // Act - try to get different selections by varying the time
+        for (int i = 0; i < 50; i++) {
+            // Use reflection to call private method for testing
+            try {
+                java.lang.reflect.Method method = EndpointRegistry.class.getDeclaredMethod("selectBootstrapServer", List.class);
```

Comment:
Rather than use reflection, better to make the method static with default access and mark @VisibleForTesting.    I'd also pass a Clock in as an argument, so you have a more elegant way to control time in your test.  The production code would just pass Clock.systemUTC().

Reply:
updated